src/components/TrackMaintaining.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, Fragment } from 'react'
+
+import { useActionableState } from '../hooks/useActionableOnly'
+
+import { useView, setOptionsInUrl } from '../hooks/useUrlState'
+import { TrackAside } from './TrackAside'
+import { SwitchToggle } from './SwitchToggle'
+import { ViewSelectLink } from './ViewSelectLink'
+import { useRemoteConfig } from '../hooks/useRemoteConfig'
+import { TrackIcon } from './TrackIcon'
+import { TrackDescription } from './TrackDescription'
+import { ExerciseDetails } from './views/ExerciseDetails'
+import { LaunchList } from './views/LaunchList'
+import { ExerciseTree } from './views/ExerciseTree'
+
+export function TrackMaintaining({
+  trackId,
+}: {
+  trackId: TrackIdentifier
+}): JSX.Element {
+  const [selectedView] = useView()
+  const actualView = (selectedView || DEFAULT_VIEW) as View
+
+  const doHideExercise = useCallback(() => {
+    // Heuristic, if there is a "back" state, go back
+    if (
+      window &&
+      window.history &&
+      window.history.state &&
+      window.history.state.previous
+    ) {
+      if (window.history.state.previous.trackId === trackId) {
+        window.history.back()
+        return
+      }
+    }
+
+    // Otherwise, hide by going to the default view
+    setOptionsInUrl({ view: DEFAULT_VIEW, exercise: '' })
```

Comment:
Sidenote: if we want a lot more pages, I think we'll just add `react-router` instead of doing it manually. This was a quick routerless thing. Make me an issue and I'll do it :)

Reply:
Done: https://github.com/exercism/v3-dashboard/issues/7